x/pep/keeper/query_decrypt_data.go

```diff
@@ -0,0 +1,116 @@
+package keeper
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+
+	enc "github.com/FairBlock/DistributedIBE/encryption"
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/drand/kyber"
+	bls "github.com/drand/kyber-bls12381"
+	"github.com/drand/kyber/pairing"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) DecryptData(goCtx context.Context, req *types.QueryDecryptDataRequest) (*types.QueryDecryptDataResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if req.Pubkey == "" {
+		pk, found := k.GetActivePubKey(ctx)
+		if !found {
+			return &types.QueryDecryptDataResponse{}, errors.New("pubkey not found")
+		}
+		req.Pubkey = pk.PublicKey
+	}
+
+	suite := bls.NewBLS12381Suite()
+	publicKeyPoint, err := k.getPubKeyPoint(req.Pubkey, suite)
+	if err != nil {
+		return &types.QueryDecryptDataResponse{}, err
+	}
+
+	skPoint, err := k.getSKPoint(req.AggrKeyshare, suite)
+	if err != nil {
+		return &types.QueryDecryptDataResponse{}, err
+	}
+
+	dataBytes, err := hex.DecodeString(req.EncryptedData)
+	if err != nil {
+		return &types.QueryDecryptDataResponse{}, err
+	}
+
+	var decryptedData bytes.Buffer
+	var dataBuffer bytes.Buffer
+	_, err = dataBuffer.Write(dataBytes)
+	if err != nil {
+		return &types.QueryDecryptDataResponse{}, err
+	}
+
+	err = enc.Decrypt(publicKeyPoint, skPoint, &decryptedData, &dataBuffer)
+	if err != nil {
+		return &types.QueryDecryptDataResponse{}, err
+	}
+
+	decodedDataString := hex.EncodeToString(decryptedData.Bytes())
+	return &types.QueryDecryptDataResponse{
+		DecryptedData: decodedDataString,
+	}, nil
+}
+
+func (k Keeper) getSKPoint(
```

Comment:
There is another `getSKPoint()` implementation in the `x/pep/module/module.go`, Maybe we can change this function to public and in the `module.go` call the function from here `am.keeper.GetSKPoint()` instead of having the same function in two separate places

Reply:
done